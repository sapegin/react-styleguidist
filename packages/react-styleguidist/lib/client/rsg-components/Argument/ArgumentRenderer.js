import "core-js/modules/es6.object.assign";
import "core-js/modules/web.dom.iterable";
import "core-js/modules/es6.array.iterator";
import "core-js/modules/es6.object.keys";
import "core-js/modules/es6.function.name";

import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Markdown from 'rsg-components/Markdown';
import Name from 'rsg-components/Name';
import Type from 'rsg-components/Type';
import Group from 'react-group';

function _extends() { _extends = Object.assign || function (target) { for (let i = 1; i < arguments.length; i++) { const source = arguments[i]; for (const key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) {return {};} const target = _objectWithoutPropertiesLoose(source, excluded); let key; let i; if (Object.getOwnPropertySymbols) { const sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) {continue;} if (!Object.prototype.propertyIsEnumerable.call(source, key)) {continue;} target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) {return {};} const target = {}; const sourceKeys = Object.keys(source); let key; let i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) {continue;} target[key] = source[key]; } return target; }
export var styles = function styles(_ref) {
  const space = _ref.space;
  return {
    block: {
      marginBottom: space[2]
    }
  };
};
export function ArgumentRenderer(_ref2) {
  const classes = _ref2.classes;

      
const name = _ref2.name;

      
const type = _ref2.type;

      
const description = _ref2.description;

      
const returns = _ref2.returns;

      
const block = _ref2.block;

      
const props = _objectWithoutProperties(_ref2, ["classes", "name", "type", "description", "returns", "block"]);

  return React.createElement(Group, _extends({
    className: block && classes.block
  }, props), returns && 'Returns', name && React.createElement("span", null, React.createElement(Name, null, name), type && ':'), type && React.createElement(Type, null, type.name), type && description && " \u2014 ", description && React.createElement(Markdown, {
    text: "".concat(description),
    inline: true
  }));
}
ArgumentRenderer.propTypes = {
  classes: PropTypes.object.isRequired,
  name: PropTypes.string,
  type: PropTypes.object,
  description: PropTypes.string,
  returns: PropTypes.bool,
  block: PropTypes.bool
};
export default Styled(styles)(ArgumentRenderer);