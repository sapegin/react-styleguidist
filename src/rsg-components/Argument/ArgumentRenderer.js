import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Markdown from 'rsg-components/Markdown';
import Code from 'rsg-components/Code';
import Name from 'rsg-components/Name';
import Group from 'react-group';

export const styles = ({ color }) => ({
	type: {
		fontSize: 'inherit',
		color: color.type,
	},
});

export function ArgumentRenderer({ classes, name, type, description, returns, ...props }) {
	return (
		<Group {...props}>
			{returns && 'Returns'}
			{name &&
				<span>
					<Name name={name} />
					{type && ':'}
				</span>}
			{type && <Code className={classes.type}>{type.name}</Code>}
			{description && <Markdown text={`â€” ${description}`} inline />}
		</Group>
	);
}

ArgumentRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	name: PropTypes.string,
	type: PropTypes.object,
	description: PropTypes.string,
	returns: PropTypes.bool,
};

export default Styled(styles)(ArgumentRenderer);
