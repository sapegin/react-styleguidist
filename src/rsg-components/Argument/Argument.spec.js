import React from 'react';
import { ArgumentRenderer } from './ArgumentRenderer';

const name = 'Foo';
const type = { name: 'Array' };
const description = 'Converts foo to bar';

it('renderer should render argument', () => {
	const actual = shallow(
		<ArgumentRenderer classes={{}} name={name} type={type} description={description} />
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render argument without type', () => {
	const actual = shallow(<ArgumentRenderer classes={{}} name={name} description={description} />);

	expect(actual).toMatchSnapshot();
});

it('renderer should render argument without description', () => {
	const actual = shallow(<ArgumentRenderer classes={{}} name={name} type={type} />);

	expect(actual).toMatchSnapshot();
});

it('renderer should render return value', () => {
	const actual = shallow(
		<ArgumentRenderer classes={{}} type={type} description={description} returns />
	);

	expect(actual).toMatchSnapshot();
});
