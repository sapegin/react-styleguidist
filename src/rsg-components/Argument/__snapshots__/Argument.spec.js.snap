// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render argument 1`] = `
<Group
  separator=" "
>
  <span>
    <Consumer(Name)>
      Foo
    </Consumer(Name)>
    :
  </span>
  <Consumer(Type)>
    Array
  </Consumer(Type)>
   — 
  <Markdown
    inline={true}
    text="Converts foo to bar"
  />
</Group>
`;

exports[`should render argument without description 1`] = `
<Group
  separator=" "
>
  <span>
    <Consumer(Name)>
      Foo
    </Consumer(Name)>
    :
  </span>
  <Consumer(Type)>
    Array
  </Consumer(Type)>
</Group>
`;

exports[`should render argument without type 1`] = `
<Group
  separator=" "
>
  <span>
    <Consumer(Name)>
      Foo
    </Consumer(Name)>
  </span>
  <Markdown
    inline={true}
    text="Converts foo to bar"
  />
</Group>
`;

exports[`should render default value of argument 1`] = `
<Group
  default="bar"
  separator=" "
>
  <Consumer(Type)>
    String
    =bar
  </Consumer(Type)>
   — 
  <Markdown
    inline={true}
    text="Converts foo to bar"
  />
</Group>
`;

exports[`should render default value of optional argument 1`] = `
<Group
  default="true"
  separator=" "
>
  <Consumer(Type)>
    Boolean
    ?
    =true
  </Consumer(Type)>
   — 
  <Markdown
    inline={true}
    text="Converts foo to bar"
  />
</Group>
`;

exports[`should render optional argument 1`] = `
<Group
  separator=" "
>
  <Consumer(Type)>
    Array
    ?
  </Consumer(Type)>
   — 
  <Markdown
    inline={true}
    text="Converts foo to bar"
  />
</Group>
`;

exports[`should render return value 1`] = `
<Group
  separator=" "
>
  Returns
  <Consumer(Type)>
    Array
  </Consumer(Type)>
   — 
  <Markdown
    inline={true}
    text="Converts foo to bar"
  />
</Group>
`;

exports[`should render with block styles 1`] = `
<Group
  className="block"
  separator=" "
/>
`;
